### DISCUSSION TITLE
AI for Data Visualisation

### DISCUSSION THREAD

**Discussion by clizbe** [TulipaEnergy/TulipaEnergyModel.jl#614] [Ideas]
Someone suggested we use an AI language model to build whatever graphs people want.
My gut says that's very black-box hand-wavy so you can't trust the results - and obviously it'd have to be something off-the-shelf so we don't have to maintain it.
But maybe for an online thing for stakeholders? Is there an off-the-shelf AI visualiser we could use?

**Comment by datejada** [comment]
It is not entirely far-fetched. I have seen an application for a model in the US. I looked for the post on LinkedIn, but I couldn't find it with a quick search. I will search more thoroughly to get the reference. The model is a Large Language Model (LLM) that is trained with a lot of numerical results and analysis. It can provide answers to questions such as 'Which are the top producers per technology X, Y, Z?' or 'Give an analysis of the hours with the highest prices, or which hours present the most congestion in the grid.' 

While you can achieve similar results with a well-crafted dashboard, the advantage of this model is that it uses natural language to access, analyze, summarize, and plot the data, whereas a dashboard is fixed to what you have defined. With this model, you can explore the results. 

Depending on how we want to explore this option, we could work with a master's student, perhaps from the mathematical department of TU Delft, or a KIP with help from people who have already worked with some LLM at TNO.

**Reply by clizbe** [reply]
I agree it would be a separate project of some kind - but it would just be the setup of an off-the-shelf solution. I don't think we should develop something ourselves.

**Comment by clizbe** [comment]
Let's dream about this later.
